signac/core/h5store.py

```diff
@@ -298,7 +311,7 @@ def open(self):
         import h5py
         self._thread_lock.acquire()
         try:
-            self._file = h5py.File(self._filename)
+            self._file = h5py.File(self._filename, mode=self.mode, **self._kwargs)
```

Comment:
Should this be `self._mode`? edit: nevermind, I see the property, but why do we expose that property in the first place?

Reply:
It should be `_mode`. The mode is a property of the `H5Store`, why not expose it?

Reply:
I see it is exposed for file objects. We can expose it. https://docs.python.org/3/library/io.html#io.FileIO.mode